views/watchlist_funcs.php

```diff
@@ -1,41 +1,31 @@
  <?php
+  require_once("../database/setup.php");
 
   // Follow or unfollow an item ()
-  if (!isset($_POST['functionname']) || !isset($_POST['arguments'])) {
+  if (!isset($_POST['functionname']) || !isset($_POST['argument_1']) || !isset($_POST['argument_2']) || !isset($_POST['argument_3'])) {
     return;
   }
+  // Extract variables from the POST variables (follower_rows exists to check that the user isn't already on the watchlist)
+  $item_id = $_POST['argument_1'];
+  $user_id = $_POST['argument_2'];
+  $follower_rows = $_POST['argument_3'];
 
-  // Extract arguments from the POST variables:
-  $item_id = $_POST['arguments'];
-
-  if (!isset($_SESSION['userId'])) {
-    $error = 'You must be logged in to add an item to your watchlist.';
-    echo  "<p><span class='font-weight-bold'>Error: </span> $error</p>";
-  } else {
-    $user_id = intval($_SESSION['userId']);
-    if ($user_id == 0) {
-      $error = 'Invalid userId.';
-      echo  "<p><span class='font-weight-bold'>Error: </span> $error</p>";
+  // Use argument to update the database (add or remove an item to the watchlist), returning success (if successful)
+  if ($_POST['functionname'] == 'add_to_watchlist') {
+    if ($follower_rows < 1) {
+      $query = "INSERT INTO FollowedItems (userId, itemId) VALUES ('$user_id', '$item_id')";
+      $db->query($query);
     }
-  }
-  // Preparing to query the database 
-  // Question: do I need this? I'd like to understand why we're being explicit about the database we're connecting to
-  $db->query("USE auction_site");
-
-  // Update the database (add or remove an item to the watchlist) and return success/failure
-  if ($_POST['functionname'] == "add_to_watchlist") {
-    $query = "INSERT INTO FollowedItems ($user_id, $item_id)";
-    $db->query($query);
-    $res = "You have successfully been added to the watchlist";
-  } else if ($_POST['functionname'] == "remove_from_watchlist") {
-    $query = "DELETE FROM FollowedItems ($user_id, $item_id)";
+    $res = "success";

```

Comment:
I would personally advise to use somehting a bit more telling "successfully added to the watchlist" or so, that way, in the other screen you can make sure that the right action was successful - I checked and this works fine, but in theory, it could happen that you actually succeeded removing but the other side thought you succeeded adding but in reality you deleted.

Reply:
I've changed it, as I agree that it does add clarity, and it makes it easier to see if you're working on the add_to or remove_from function. However, as far as I can tell, you don't actually see any of these messages, the computer just checks if both strings are the same